sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3985,7 +4073,15 @@ pi_result piextQueueGetNativeHandle(pi_queue Queue,
 
   // Extract a Level Zero compute queue handle from the given PI queue
   uint32_t QueueGroupOrdinalUnused;
-  *ZeQueue = Queue->ComputeQueueGroup.getZeQueue(&QueueGroupOrdinalUnused);
+  auto TID = std::this_thread::get_id();
+  auto InitialGroup = Queue->ComputeQueueGroupsByTID.begin()->second;
+  std::pair<std::unordered_map<std::thread::id,
+                               _pi_queue::pi_queue_group_t>::iterator,
+            bool>
+      Result = Queue->ComputeQueueGroupsByTID.insert({TID, InitialGroup});
```

Comment:
```suggestion
  auto Result = Queue->ComputeQueueGroupsByTID.insert({TID, InitialGroup});
```
A more important Q is why you create a new queue group in the API that is supposed to extract an existing native queue group? (not sure for which TID though) At the very list you should check if a group exists for the current TID and return that, no?

Reply:
The insert is a conditional insert, if existing entry with same key does not exist.

Reply:
Indeed, OK.